server/chroma/src/routes/authorization.rs

```diff
@@ -38,6 +38,23 @@ impl FromRequest for Authorization {
                     trace!("Value of authorization header could not be converter to UTF-8 String")
                 })?;
 
+            // Check if we're dealing with a service token
+            if authorization.starts_with("Service ") {
+                let token = authorization.chars().skip(8).collect::<String>();
+                if token.is_empty() {
+                    return Err(AuthorizationError::InvalidServiceToken);
+                }
+
+                return if data.config.service_tokens().contains(&token.as_str()) {
```

Comment:
Using `contains()` means the token can contain extra characters. I think we want a stricter equality here (after maybe stripping trailing whitespace).

Also note that most standard string comparison functions are susceptible to timing attacks. That shouldn't be a problem if we have some good rate limits in place, but it's something to double check.

Reply:
Note that contains() in this case acts on a `Vec`, rather than a `String`. The `service_tokens()` function splits out the configured `service_tokens` configuration variable by the `,` delimiter.

Timing attacks could be an issue. I thought I had added a ratelimiter, apparently I hadn't. Will add that too.

Reply:
313c479 adds a ratelimiter with a limit of 3 per second, and a maximum burst of 10 requests.  
[actix_governor](https://docs.rs/actix-governor/latest/actix_governor/#)

Reply:
Ah indeed, I was reading wrong. Thanks for adding the rate limiter!